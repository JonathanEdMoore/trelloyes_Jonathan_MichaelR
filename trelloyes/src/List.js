import React from 'react';

import './styles/List.css';

import Card from './Card';

function List(props) {
  console.log(props);
  console.log(props.cards);
  const cards = props.cards.map((card, i) => {
    return(
      <Card key={i} title={card.title} content={card.content}/>
    )
  }); 
  return (
    <section className="List">
      <header className="List-header">
        <h2>{props.header}</h2>
      </header>
      <div className="List-cards">
        {cards}
        <button type="button" className="List-add-button">+ Add Random Card</button>
      </div>
    </section>
  );
};

export default List;