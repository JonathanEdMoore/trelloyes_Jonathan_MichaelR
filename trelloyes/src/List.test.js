import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

it('renders without crashing', () =>{
  const div = document.createElement('div');
  ReactDOM.render(<List header="MyHeader" cards={[]}/> , div);
  ReactDOM.unmountComponentAtNode(div);
})

//Snapshot Tests
describe('<List />', () => {
  const snap = renderer.create(<List header="MyHeader" cards={[]}/>).toJSON();
  expect(snap).toMatchSnapshot();
})