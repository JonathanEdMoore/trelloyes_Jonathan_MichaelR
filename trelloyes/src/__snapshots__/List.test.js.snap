// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      MyHeader
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
